<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "https://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.17"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>SharpNEAT: Class List</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="navtree.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="resize.js"></script>
<script type="text/javascript" src="navtreedata.js"></script>
<script type="text/javascript" src="navtree.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/searchdata.js"></script>
<script type="text/javascript" src="search/search.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">SharpNEAT
   </div>
   <div id="projectbrief">C# implementation of NEAT</div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.17 -->
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
var searchBox = new SearchBox("searchBox", "search",false,'Search');
/* @license-end */
</script>
<script type="text/javascript" src="menudata.js"></script>
<script type="text/javascript" src="menu.js"></script>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
$(function() {
  initMenu('',true,false,'search.php','Search');
  $(document).ready(function() { init_search(); });
});
/* @license-end */</script>
<div id="main-nav"></div>
</div><!-- top -->
<div id="side-nav" class="ui-resizable side-nav-resizable">
  <div id="nav-tree">
    <div id="nav-tree-contents">
      <div id="nav-sync" class="sync"></div>
    </div>
  </div>
  <div id="splitbar" style="-moz-user-select:none;" 
       class="ui-resizable-handle">
  </div>
</div>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
$(document).ready(function(){initNavTree('annotated.html',''); initResizable(); });
/* @license-end */
</script>
<div id="doc-content">
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div class="header">
  <div class="headertitle">
<div class="title">Class List</div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock">Here are the classes, structs, unions and interfaces with brief descriptions:</div><div class="directory">
<div class="levels">[detail level <span onclick="javascript:toggleLevel(1);">1</span><span onclick="javascript:toggleLevel(2);">2</span><span onclick="javascript:toggleLevel(3);">3</span><span onclick="javascript:toggleLevel(4);">4</span>]</div><table class="directory">
<tr id="row_0_" class="even"><td class="entry"><span style="width:0px;display:inline-block;">&#160;</span><span id="arr_0_" class="arrow" onclick="toggleFolder('0_')">&#9660;</span><span class="icona"><span class="icon">N</span></span><a class="el" href="namespace_sharp_neat.html" target="_self">SharpNeat</a></td><td class="desc"></td></tr>
<tr id="row_0_0_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_0_0_" class="arrow" onclick="toggleFolder('0_0_')">&#9658;</span><span class="icona"><span class="icon">N</span></span><a class="el" href="namespace_sharp_neat_1_1_core.html" target="_self">Core</a></td><td class="desc"></td></tr>
<tr id="row_0_0_0_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="struct_sharp_neat_1_1_core_1_1_aux_fitness_info.html" target="_self">AuxFitnessInfo</a></td><td class="desc">Auxiliary fitness info, i.e. for evaluation metrics other than the primary fitness metric but that nonetheless we are interested in observing </td></tr>
<tr id="row_0_0_1_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_core_1_1_coordinate_vector.html" target="_self">CoordinateVector</a></td><td class="desc">General purpose representation of a point in a multidimensional space. A vector of coordinates, each coordinate defining the position within a dimension/axis defined by an ID </td></tr>
<tr id="row_0_0_2_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_core_1_1_evaluation_info.html" target="_self">EvaluationInfo</a></td><td class="desc">Encapsulates genome evaluation data; specifically whether the genome has been evaluated, the genome's fitness, how many times the genome has been evaluated and if more than once then the mean fitness </td></tr>
<tr id="row_0_0_3_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="struct_sharp_neat_1_1_core_1_1_fitness_info.html" target="_self">FitnessInfo</a></td><td class="desc">Wrapper struct for fitness values </td></tr>
<tr id="row_0_0_4_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_core_1_1_genome_fitness_comparer.html" target="_self">GenomeFitnessComparer</a></td><td class="desc">Sort genomes, highest fitness first. <a class="el" href="namespace_sharp_neat_1_1_genomes.html">Genomes</a> with equal fitness are secondary sorted by age (youngest first). Used by the selection routines to select the fittest and youngest genomes </td></tr>
<tr id="row_0_0_5_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="interface_sharp_neat_1_1_core_1_1_i_distance_metric.html" target="_self">IDistanceMetric</a></td><td class="desc">Represents a metric for measuring the distance between two genome positions in an encoding space, and thus the compatibility of the two genomes with respect to the probability of creating fit offspring </td></tr>
<tr id="row_0_0_6_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="interface_sharp_neat_1_1_core_1_1_i_evolution_algorithm.html" target="_self">IEvolutionAlgorithm</a></td><td class="desc">A generic interface for evolution algorithm classes </td></tr>
<tr id="row_0_0_7_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="interface_sharp_neat_1_1_core_1_1_i_genome.html" target="_self">IGenome</a></td><td class="desc">Generic interface for genome classes </td></tr>
<tr id="row_0_0_8_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="interface_sharp_neat_1_1_core_1_1_i_genome_decoder.html" target="_self">IGenomeDecoder</a></td><td class="desc">Generic interface for classes that decode genomes into phenomes </td></tr>
<tr id="row_0_0_9_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="interface_sharp_neat_1_1_core_1_1_i_genome_factory.html" target="_self">IGenomeFactory</a></td><td class="desc">Generic interface genome factory classes. Genome factories are intended to be used for creating initial populations, either random populations or from one or more seed genomes. However, genome reproduction is performed via the overloaded <a class="el" href="interface_sharp_neat_1_1_core_1_1_i_genome.html#aaa6484beaf3243a4759c72565dc3f8f0" title="Asexual reproduction.">IGenome.CreateOffspring()</a> methods, placing these methods on <a class="el" href="interface_sharp_neat_1_1_core_1_1_i_genome.html" title="Generic interface for genome classes.">IGenome</a> gives reproduction methods direct and convenient access to internal genome data </td></tr>
<tr id="row_0_0_10_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="interface_sharp_neat_1_1_core_1_1_i_genome_list_evaluator.html" target="_self">IGenomeListEvaluator</a></td><td class="desc">Generic interface for evaluating a list of genomes. By operating on a list we allow concrete implementations of this interface to choose between evaluating each genome independently of the others, perhaps across several execution threads, or in some collective evaluation scheme such as an artificial life/world scenario </td></tr>
<tr id="row_0_0_11_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="interface_sharp_neat_1_1_core_1_1_i_phenome_evaluator.html" target="_self">IPhenomeEvaluator</a></td><td class="desc">Generic interface for phenome evaluation classes. Evaluates and assigns a fitness to individual TPhenome's </td></tr>
<tr id="row_0_0_12_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="interface_sharp_neat_1_1_core_1_1_i_speciation_strategy.html" target="_self">ISpeciationStrategy</a></td><td class="desc">Represents a strategy for dividing genomes into distinct species </td></tr>
<tr id="row_0_0_13_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_core_1_1_parallel_genome_list_evaluator.html" target="_self">ParallelGenomeListEvaluator</a></td><td class="desc">A concrete implementation of <a class="el" href="interface_sharp_neat_1_1_core_1_1_i_genome_list_evaluator.html" title="Generic interface for evaluating a list of genomes. By operating on a list we allow concrete implemen...">IGenomeListEvaluator</a> that evaluates genomes independently of each other and in parallel (on multiple execution threads) </td></tr>
<tr id="row_0_0_14_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_core_1_1_selective_genome_list_evaluator.html" target="_self">SelectiveGenomeListEvaluator</a></td><td class="desc">An <a class="el" href="interface_sharp_neat_1_1_core_1_1_i_genome_list_evaluator.html" title="Generic interface for evaluating a list of genomes. By operating on a list we allow concrete implemen...">IGenomeListEvaluator</a> that wraps another <a class="el" href="interface_sharp_neat_1_1_core_1_1_i_genome_list_evaluator.html" title="Generic interface for evaluating a list of genomes. By operating on a list we allow concrete implemen...">IGenomeListEvaluator</a> and filters/selects the genomes that are to be passed to the wrapped evaluator based on some predicate/test </td></tr>
<tr id="row_0_0_15_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_core_1_1_serial_genome_list_evaluator.html" target="_self">SerialGenomeListEvaluator</a></td><td class="desc">A concrete implementation of <a class="el" href="interface_sharp_neat_1_1_core_1_1_i_genome_list_evaluator.html" title="Generic interface for evaluating a list of genomes. By operating on a list we allow concrete implemen...">IGenomeListEvaluator</a> that evaluates genomes independently of each other and in series on a single thread </td></tr>
<tr id="row_0_0_16_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_core_1_1_specie.html" target="_self">Specie</a></td><td class="desc">Represents a single specie within a speciated population </td></tr>
<tr id="row_0_0_17_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_core_1_1_update_scheme.html" target="_self">UpdateScheme</a></td><td class="desc">Represents an update scheme for an <a class="el" href="interface_sharp_neat_1_1_core_1_1_i_evolution_algorithm.html" title="A generic interface for evolution algorithm classes.">IEvolutionAlgorithm</a>. e.g. update per some time duration or some number of generations </td></tr>
<tr id="row_0_1_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_0_1_" class="arrow" onclick="toggleFolder('0_1_')">&#9658;</span><span class="icona"><span class="icon">N</span></span><a class="el" href="namespace_sharp_neat_1_1_decoders.html" target="_self">Decoders</a></td><td class="desc"></td></tr>
<tr id="row_0_1_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_0_1_0_" class="arrow" onclick="toggleFolder('0_1_0_')">&#9658;</span><span class="icona"><span class="icon">N</span></span><a class="el" href="namespace_sharp_neat_1_1_decoders_1_1_hyper_neat.html" target="_self">HyperNeat</a></td><td class="desc"></td></tr>
<tr id="row_0_1_0_0_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_decoders_1_1_hyper_neat_1_1_default_node_set_mapping_function.html" target="_self">DefaultNodeSetMappingFunction</a></td><td class="desc">Defines a mapping between two node sets based on mapping all source nodes to all target nodes, but with an option to omit mappings where the distance between source and target node is over some threshold </td></tr>
<tr id="row_0_1_0_1_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_decoders_1_1_hyper_neat_1_1_hyper_neat_decoder.html" target="_self">HyperNeatDecoder</a></td><td class="desc">Decodes CPPN NeatGenome's into concrete network instances. This decoder uses a HyperNEAT substrate and queries a CPPN NEAT network to generate/grow a network from the substrate </td></tr>
<tr id="row_0_1_0_2_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="interface_sharp_neat_1_1_decoders_1_1_hyper_neat_1_1_i_node_set_mapping_function.html" target="_self">INodeSetMappingFunction</a></td><td class="desc">Implementations of <a class="el" href="interface_sharp_neat_1_1_decoders_1_1_hyper_neat_1_1_i_node_set_mapping_function.html" title="Implementations of INodeSetMappingFunction define a mapping between source and target node sets....">INodeSetMappingFunction</a> define a mapping between source and target node sets. Each mapping is interpreted/used as a connection between the source and target node. E.g. the simplest such function is to map every source node to every target node (N*M connections) </td></tr>
<tr id="row_0_1_0_3_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_decoders_1_1_hyper_neat_1_1_node_set_mapping.html" target="_self">NodeSetMapping</a></td><td class="desc">Describes a mapping between nodesets. Packages an <a class="el" href="interface_sharp_neat_1_1_decoders_1_1_hyper_neat_1_1_i_node_set_mapping_function.html" title="Implementations of INodeSetMappingFunction define a mapping between source and target node sets....">INodeSetMappingFunction</a> with indexes into a list of node sets that identify the source and target nodesets for the mapping </td></tr>
<tr id="row_0_1_0_4_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_decoders_1_1_hyper_neat_1_1_substrate.html" target="_self">Substrate</a></td><td class="desc">HyperNEAT substrate. Encapsulates substrate nodes in sets and connections. Connections can be defined explicitly or by providing mapping functions that map (connect) between nodes in sets. Node sets can be arranged as layers, however there is no limitation on node positions within the substrate - nodes in a set can be distributed throughout the substrate with no restrictions based on e.g. where nodes in other sets are located </td></tr>
<tr id="row_0_1_0_5_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_decoders_1_1_hyper_neat_1_1_substrate_connection.html" target="_self">SubstrateConnection</a></td><td class="desc">Represents a connection between two nodes in a HyperNEAT substrate. The node positions are represented as arrays of numbers so as not to limit the number of dimensions that positions (and therefore substrates) can be defined within </td></tr>
<tr id="row_0_1_0_6_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="struct_sharp_neat_1_1_decoders_1_1_hyper_neat_1_1_substrate_node.html" target="_self">SubstrateNode</a></td><td class="desc">Represents a node within a HyperNEAT substrate </td></tr>
<tr id="row_0_1_0_7_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_decoders_1_1_hyper_neat_1_1_substrate_node_set.html" target="_self">SubstrateNodeSet</a></td><td class="desc">Represents a set of nodes on a substrate. Nodesets are used to represent sets such as the input and output nodes. Hidden nodes can be represented as layers with each layer being represented by a set. This allows connection mapping to be defined between sets </td></tr>
<tr id="row_0_1_1_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_0_1_1_" class="arrow" onclick="toggleFolder('0_1_1_')">&#9658;</span><span class="icona"><span class="icon">N</span></span><a class="el" href="namespace_sharp_neat_1_1_decoders_1_1_neat.html" target="_self">Neat</a></td><td class="desc"></td></tr>
<tr id="row_0_1_1_0_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_decoders_1_1_neat_1_1_neat_genome_decoder.html" target="_self">NeatGenomeDecoder</a></td><td class="desc">Decodes NeatGenome's into concrete network instances </td></tr>
<tr id="row_0_1_2_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_decoders_1_1_cyclic_network_factory.html" target="_self">CyclicNetworkFactory</a></td><td class="desc">Static factory for creating CyclicNetwork's from INetworkDefinition's </td></tr>
<tr id="row_0_1_3_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_decoders_1_1_fast_acyclic_network_factory.html" target="_self">FastAcyclicNetworkFactory</a></td><td class="desc">Static factory for creating AcyclicNetwork(s) from INetworkDefinition(s) </td></tr>
<tr id="row_0_1_4_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_decoders_1_1_fast_cyclic_network_factory.html" target="_self">FastCyclicNetworkFactory</a></td><td class="desc">Static factory for creating CyclicNetwork's from INetworkDefinition's </td></tr>
<tr id="row_0_1_5_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_decoders_1_1_network_activation_scheme.html" target="_self">NetworkActivationScheme</a></td><td class="desc">Represents network activation schemes. E.g. fixed number of activation timesteps or activation until the network becomes 'relaxed'. Relaxed here means that no node's output value changed by more than some threshold value </td></tr>
<tr id="row_0_2_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_0_2_" class="arrow" onclick="toggleFolder('0_2_')">&#9658;</span><span class="icona"><span class="icon">N</span></span><a class="el" href="namespace_sharp_neat_1_1_distance_metrics.html" target="_self">DistanceMetrics</a></td><td class="desc"></td></tr>
<tr id="row_0_2_0_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_distance_metrics_1_1_distance_metric_utils.html" target="_self">DistanceMetricUtils</a></td><td class="desc">Static helper methods for distance metrics </td></tr>
<tr id="row_0_2_1_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_distance_metrics_1_1_euclidean_distance_metric.html" target="_self">EuclideanDistanceMetric</a></td><td class="desc">Euclidean distance metric </td></tr>
<tr id="row_0_2_2_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_distance_metrics_1_1_manhattan_distance_metric.html" target="_self">ManhattanDistanceMetric</a></td><td class="desc">Manhattan distance metric </td></tr>
<tr id="row_0_3_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_0_3_" class="arrow" onclick="toggleFolder('0_3_')">&#9658;</span><span class="icona"><span class="icon">N</span></span><a class="el" href="namespace_sharp_neat_1_1_evolution_algorithms.html" target="_self">EvolutionAlgorithms</a></td><td class="desc"></td></tr>
<tr id="row_0_3_0_" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_0_3_0_" class="arrow" onclick="toggleFolder('0_3_0_')">&#9658;</span><span class="icona"><span class="icon">N</span></span><a class="el" href="namespace_sharp_neat_1_1_evolution_algorithms_1_1_complexity_regulation.html" target="_self">ComplexityRegulation</a></td><td class="desc"></td></tr>
<tr id="row_0_3_0_0_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_evolution_algorithms_1_1_complexity_regulation_1_1_default_complexity_regulation_strategy.html" target="_self">DefaultComplexityRegulationStrategy</a></td><td class="desc">Default complexity regulation strategy. This strategy provides a choice of hard/absolute complexity ceiling or a ceiling relative to the complexity at the end of the most recent simplification phase. The strategy transitions from complexifying to simplifying when the ceiling is reached. Transitioning from simplifying to complexifying occurs when complexity is no longer falling <em>and</em> complexity is below the ceiling. This is determined by tracking a complexity moving average calculated over the past N generations </td></tr>
<tr id="row_0_3_0_1_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="interface_sharp_neat_1_1_evolution_algorithms_1_1_complexity_regulation_1_1_i_complexity_regulation_strategy.html" target="_self">IComplexityRegulationStrategy</a></td><td class="desc">Represents a complexity regulation strategy </td></tr>
<tr id="row_0_3_0_2_" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_evolution_algorithms_1_1_complexity_regulation_1_1_null_complexity_regulation_strategy.html" target="_self">NullComplexityRegulationStrategy</a></td><td class="desc">Null strategy. Fixed to Complexifying mode </td></tr>
<tr id="row_0_3_1_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_evolution_algorithms_1_1_abstract_generational_algorithm.html" target="_self">AbstractGenerationalAlgorithm</a></td><td class="desc">Abstract class providing some common/baseline data and methods for implementations of IEvolutionAlgorithm </td></tr>
<tr id="row_0_3_2_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_evolution_algorithms_1_1_neat_algorithm_stats.html" target="_self">NeatAlgorithmStats</a></td><td class="desc">Neat evolution algorithm statistics </td></tr>
<tr id="row_0_3_3_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_evolution_algorithms_1_1_neat_evolution_algorithm.html" target="_self">NeatEvolutionAlgorithm</a></td><td class="desc">Implementation of the NEAT evolution algorithm. Incorporates: </td></tr>
<tr id="row_0_3_4_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_evolution_algorithms_1_1_neat_evolution_algorithm_parameters.html" target="_self">NeatEvolutionAlgorithmParameters</a></td><td class="desc">Parameters specific to the NEAT evolution algorithm </td></tr>
<tr id="row_0_4_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_0_4_" class="arrow" onclick="toggleFolder('0_4_')">&#9658;</span><span class="icona"><span class="icon">N</span></span><a class="el" href="namespace_sharp_neat_1_1_genomes.html" target="_self">Genomes</a></td><td class="desc"></td></tr>
<tr id="row_0_4_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_0_4_0_" class="arrow" onclick="toggleFolder('0_4_0_')">&#9658;</span><span class="icona"><span class="icon">N</span></span><a class="el" href="namespace_sharp_neat_1_1_genomes_1_1_hyper_neat.html" target="_self">HyperNeat</a></td><td class="desc"></td></tr>
<tr id="row_0_4_0_0_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_genomes_1_1_hyper_neat_1_1_cppn_genome_factory.html" target="_self">CppnGenomeFactory</a></td><td class="desc">A sub-class of NeatGenomeFactory for creating CPPN genomes </td></tr>
<tr id="row_0_4_1_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_0_4_1_" class="arrow" onclick="toggleFolder('0_4_1_')">&#9658;</span><span class="icona"><span class="icon">N</span></span><a class="el" href="namespace_sharp_neat_1_1_genomes_1_1_neat.html" target="_self">Neat</a></td><td class="desc"></td></tr>
<tr id="row_0_4_1_0_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="struct_sharp_neat_1_1_genomes_1_1_neat_1_1_added_neuron_gene_struct.html" target="_self">AddedNeuronGeneStruct</a></td><td class="desc">Represents an added neuron. When a neuron is added to a neural network in NEAT an existing connection between two neurons is discarded and replaced with the new neuron and two new connections, one connection between the source neuron and the new neuron and another from the new neuron to the target neuron. This struct represents those three IDs </td></tr>
<tr id="row_0_4_1_1_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_genomes_1_1_neat_1_1_connection_gene.html" target="_self">ConnectionGene</a></td><td class="desc">A gene that represents a single connection between neurons in NEAT </td></tr>
<tr id="row_0_4_1_2_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_genomes_1_1_neat_1_1_connection_gene_comparer.html" target="_self">ConnectionGeneComparer</a></td><td class="desc">Compares the innovation ID of ConnectionGenes </td></tr>
<tr id="row_0_4_1_3_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_genomes_1_1_neat_1_1_connection_gene_list.html" target="_self">ConnectionGeneList</a></td><td class="desc">Represents a sorted list of <a class="el" href="class_sharp_neat_1_1_genomes_1_1_neat_1_1_connection_gene.html" title="A gene that represents a single connection between neurons in NEAT.">ConnectionGene</a> objects. The sorting of the items is done on request rather than being strictly enforced at all times (e.g. as part of adding and removing genes). This approach is currently more convenient for use in some of the routines that work with NEAT genomes </td></tr>
<tr id="row_0_4_1_4_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_genomes_1_1_neat_1_1_connection_gene_list_builder.html" target="_self">ConnectionGeneListBuilder</a></td><td class="desc">Used for building a list of connection genes </td></tr>
<tr id="row_0_4_1_5_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_genomes_1_1_neat_1_1_connection_mutation_info.html" target="_self">ConnectionMutationInfo</a></td><td class="desc">Each instance of <a class="el" href="class_sharp_neat_1_1_genomes_1_1_neat_1_1_connection_mutation_info.html" title="Each instance of ConnectionMutationInfo describes a type of mutation and associated parameters.">ConnectionMutationInfo</a> describes a type of mutation and associated parameters </td></tr>
<tr id="row_0_4_1_6_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_genomes_1_1_neat_1_1_connection_mutation_info_list.html" target="_self">ConnectionMutationInfoList</a></td><td class="desc">Represents a list of <a class="el" href="class_sharp_neat_1_1_genomes_1_1_neat_1_1_connection_mutation_info.html" title="Each instance of ConnectionMutationInfo describes a type of mutation and associated parameters.">ConnectionMutationInfo</a> objects </td></tr>
<tr id="row_0_4_1_7_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_genomes_1_1_neat_1_1_correlation_item.html" target="_self">CorrelationItem</a></td><td class="desc">A single comparison item resulting from the comparison of two genomes. If the CorrelationItemType is Match then both connection gene properties will be non-null, otherwise one of them will be null and the other will hold a reference to a disjoint or excess connection gene </td></tr>
<tr id="row_0_4_1_8_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_genomes_1_1_neat_1_1_correlation_results.html" target="_self">CorrelationResults</a></td><td class="desc">The results from comparing two NEAT genomes and correlating their connection genes </td></tr>
<tr id="row_0_4_1_9_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_genomes_1_1_neat_1_1_correlation_statistics.html" target="_self">CorrelationStatistics</a></td><td class="desc">Statistics resulting from the comparison of two NEAT genomes </td></tr>
<tr id="row_0_4_1_10_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_genomes_1_1_neat_1_1_neat_genome.html" target="_self">NeatGenome</a></td><td class="desc">A genome class for Neuro Evolution of Augmenting Topologies (NEAT) </td></tr>
<tr id="row_0_4_1_11_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_genomes_1_1_neat_1_1_neat_genome_factory.html" target="_self">NeatGenomeFactory</a></td><td class="desc">An IGenomeFactory for NeatGenomes. We use the factory as a means of generating an initial population either randomly or using a seed genome or genomes. Subsequently all <a class="el" href="class_sharp_neat_1_1_genomes_1_1_neat_1_1_neat_genome.html" title="A genome class for Neuro Evolution of Augmenting Topologies (NEAT).">NeatGenome</a> objects keep a reference to this factory object for convenient access to <a class="el" href="class_sharp_neat_1_1_genomes_1_1_neat_1_1_neat_genome.html" title="A genome class for Neuro Evolution of Augmenting Topologies (NEAT).">NeatGenome</a> parameters and ID generator objects </td></tr>
<tr id="row_0_4_1_12_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_genomes_1_1_neat_1_1_neat_genome_parameters.html" target="_self">NeatGenomeParameters</a></td><td class="desc">Represents parameters specific to NEAT genomes. E.g. parameters that describe probabilities for the different types of mutation and the proportion of possible connections to instantiate between input and output neurons within the initial population </td></tr>
<tr id="row_0_4_1_13_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_genomes_1_1_neat_1_1_neat_genome_stats.html" target="_self">NeatGenomeStats</a></td><td class="desc">Various statistics for <a class="el" href="class_sharp_neat_1_1_genomes_1_1_neat_1_1_neat_genome.html" title="A genome class for Neuro Evolution of Augmenting Topologies (NEAT).">NeatGenome</a> </td></tr>
<tr id="row_0_4_1_14_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_genomes_1_1_neat_1_1_neuron_gene.html" target="_self">NeuronGene</a></td><td class="desc">A gene that represents a single neuron in NEAT </td></tr>
<tr id="row_0_4_1_15_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_genomes_1_1_neat_1_1_neuron_gene_list.html" target="_self">NeuronGeneList</a></td><td class="desc">Represents a sorted list of <a class="el" href="class_sharp_neat_1_1_genomes_1_1_neat_1_1_neuron_gene.html" title="A gene that represents a single neuron in NEAT.">NeuronGene</a> objects. The sorting of the items is done on request rather than being strictly enforced at all times (e.g. as part of adding and removing items). This approach is currently more convenient for use in some of the routines that work with NEAT genomes </td></tr>
<tr id="row_0_4_2_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_0_4_2_" class="arrow" onclick="toggleFolder('0_4_2_')">&#9658;</span><span class="icona"><span class="icon">N</span></span><a class="el" href="namespace_sharp_neat_1_1_genomes_1_1_rbf_neat.html" target="_self">RbfNeat</a></td><td class="desc"></td></tr>
<tr id="row_0_4_2_0_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_genomes_1_1_rbf_neat_1_1_rbf_genome_factory.html" target="_self">RbfGenomeFactory</a></td><td class="desc">A sub-class of NeatGenomeFactory for creating RBF-Neat genomes. RBF = Radial Basis Functions </td></tr>
<tr id="row_0_5_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_0_5_" class="arrow" onclick="toggleFolder('0_5_')">&#9658;</span><span class="icona"><span class="icon">N</span></span><a class="el" href="namespace_sharp_neat_1_1_network.html" target="_self">Network</a></td><td class="desc"></td></tr>
<tr id="row_0_5_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="struct_sharp_neat_1_1_network_1_1_activation_function_info.html" target="_self">ActivationFunctionInfo</a></td><td class="desc">Represents a single item in an <a class="el" href="interface_sharp_neat_1_1_network_1_1_i_activation_function_library.html" title="Represents a library of activation functions. Primarily for use in HyperNEAT.">IActivationFunctionLibrary</a>. The item represents an <a class="el" href="interface_sharp_neat_1_1_network_1_1_i_activation_function.html" title="Interface for neural network activation functions. An activation function simply takes a single input...">IActivationFunction</a> and its ID and selection probability within the owning <a class="el" href="interface_sharp_neat_1_1_network_1_1_i_activation_function_library.html" title="Represents a library of activation functions. Primarily for use in HyperNEAT.">IActivationFunctionLibrary</a> </td></tr>
<tr id="row_0_5_1_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_network_1_1_acyclic_network_depth_analysis.html" target="_self">AcyclicNetworkDepthAnalysis</a></td><td class="desc">An algorithm for analysing acyclic networks and calculating the depth of each node in the network. Bias and input nodes are defined as being at depth 0, the depth of all other nodes is defined as the maximum number of hops from the depth 0 nodes, so where multiple paths exist to a node (potentially with different numbers of hops) we take the maximum number of hops as that node's depth </td></tr>
<tr id="row_0_5_2_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_network_1_1_arc_sin_h.html" target="_self">ArcSinH</a></td><td class="desc">Rectified linear activation unit (<a class="el" href="class_sharp_neat_1_1_network_1_1_re_l_u.html" title="Rectified linear activation unit (ReLU).">ReLU</a>) </td></tr>
<tr id="row_0_5_3_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_network_1_1_arc_tan.html" target="_self">ArcTan</a></td><td class="desc"></td></tr>
<tr id="row_0_5_4_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_network_1_1_bipolar_gaussian.html" target="_self">BipolarGaussian</a></td><td class="desc">Bipolar <a class="el" href="class_sharp_neat_1_1_network_1_1_gaussian.html" title="Gaussian activation function. Output range is 0 to 1, that is, the tails of the Gaussian distribution...">Gaussian</a> activation function. Output range is -1 to 1, that is, the tails of the <a class="el" href="class_sharp_neat_1_1_network_1_1_gaussian.html" title="Gaussian activation function. Output range is 0 to 1, that is, the tails of the Gaussian distribution...">Gaussian</a> distribution curve tend towards -1 as abs(x) -&gt; Infinity and the <a class="el" href="class_sharp_neat_1_1_network_1_1_gaussian.html" title="Gaussian activation function. Output range is 0 to 1, that is, the tails of the Gaussian distribution...">Gaussian</a> peak is at y = 1 </td></tr>
<tr id="row_0_5_5_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_network_1_1_bipolar_sigmoid.html" target="_self">BipolarSigmoid</a></td><td class="desc">Bipolar sigmoid activation function. Output range is -1 to 1 instead of the more normal 0 to 1 </td></tr>
<tr id="row_0_5_6_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="struct_sharp_neat_1_1_network_1_1_connection_endpoints_struct.html" target="_self">ConnectionEndpointsStruct</a></td><td class="desc">Represents a connection between two nodes. Used primarily as a key into a Dictionary that uniquely identifies connections by their end points </td></tr>
<tr id="row_0_5_7_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_network_1_1_connection_list.html" target="_self">ConnectionList</a></td><td class="desc">A concrete implementation of <a class="el" href="interface_sharp_neat_1_1_network_1_1_i_connection_list.html" title="Represents a list of INetworkConnection&#39;s. Part of the INetworkDefinition type hierarchy.">IConnectionList</a>. Part of the <a class="el" href="interface_sharp_neat_1_1_network_1_1_i_network_definition.html" title="Abstracted representation of a network definition. This interface and the related types INodeList,...">INetworkDefinition</a> type hierarchy </td></tr>
<tr id="row_0_5_8_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_network_1_1_cyclic_network_depth_analysis.html" target="_self">CyclicNetworkDepthAnalysis</a></td><td class="desc">An algorithm for analysing cyclic networks and calculating the depth of each node in the network. Bias and input nodes are defined as being at depth 0, the depth of all other nodes is defined as the maximum number of hops from the depth 0 nodes, so where multiple paths exist to a node (potentially with different numbers of hops) we take the maximum number of hops as that node's depth </td></tr>
<tr id="row_0_5_9_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_network_1_1_cyclic_network_test.html" target="_self">CyclicNetworkTest</a></td><td class="desc">An algorithm for testing for the presence of at least one connectivity cycle within a network </td></tr>
<tr id="row_0_5_10_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_network_1_1_default_activation_function_library.html" target="_self">DefaultActivationFunctionLibrary</a></td><td class="desc">Default implementation of an <a class="el" href="interface_sharp_neat_1_1_network_1_1_i_activation_function_library.html" title="Represents a library of activation functions. Primarily for use in HyperNEAT.">IActivationFunctionLibrary</a>. Also provides static factory methods to create libraries with commonly used activation functions </td></tr>
<tr id="row_0_5_11_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_network_1_1_gaussian.html" target="_self">Gaussian</a></td><td class="desc"><a class="el" href="class_sharp_neat_1_1_network_1_1_gaussian.html" title="Gaussian activation function. Output range is 0 to 1, that is, the tails of the Gaussian distribution...">Gaussian</a> activation function. Output range is 0 to 1, that is, the tails of the <a class="el" href="class_sharp_neat_1_1_network_1_1_gaussian.html" title="Gaussian activation function. Output range is 0 to 1, that is, the tails of the Gaussian distribution...">Gaussian</a> distribution curve tend towards 0 as abs(x) -&gt; Infinity and the <a class="el" href="class_sharp_neat_1_1_network_1_1_gaussian.html" title="Gaussian activation function. Output range is 0 to 1, that is, the tails of the Gaussian distribution...">Gaussian</a> peak is at x = 0 </td></tr>
<tr id="row_0_5_12_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="interface_sharp_neat_1_1_network_1_1_i_activation_function.html" target="_self">IActivationFunction</a></td><td class="desc">Interface for neural network activation functions. An activation function simply takes a single input value and produces a single output value. <a class="el" href="interface_sharp_neat_1_1_network_1_1_i_activation_function.html" title="Interface for neural network activation functions. An activation function simply takes a single input...">IActivationFunction</a> allows for activation functions to be plugged in to neural network implementations. Typical activation functions would be a sigmoid or step function </td></tr>
<tr id="row_0_5_13_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="interface_sharp_neat_1_1_network_1_1_i_activation_function_library.html" target="_self">IActivationFunctionLibrary</a></td><td class="desc">Represents a library of activation functions. Primarily for use in HyperNEAT </td></tr>
<tr id="row_0_5_14_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="interface_sharp_neat_1_1_network_1_1_i_connection_list.html" target="_self">IConnectionList</a></td><td class="desc">Represents a list of <a class="el" href="interface_sharp_neat_1_1_network_1_1_i_network_connection.html" title="Represents a single network connection. Part of the INetworkDefinition type hierarchy.">INetworkConnection</a>'s. Part of the <a class="el" href="interface_sharp_neat_1_1_network_1_1_i_network_definition.html" title="Abstracted representation of a network definition. This interface and the related types INodeList,...">INetworkDefinition</a> type hierarchy </td></tr>
<tr id="row_0_5_15_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="interface_sharp_neat_1_1_network_1_1_i_network_connection.html" target="_self">INetworkConnection</a></td><td class="desc">Represents a single network connection. Part of the <a class="el" href="interface_sharp_neat_1_1_network_1_1_i_network_definition.html" title="Abstracted representation of a network definition. This interface and the related types INodeList,...">INetworkDefinition</a> type hierarchy </td></tr>
<tr id="row_0_5_16_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="interface_sharp_neat_1_1_network_1_1_i_network_definition.html" target="_self">INetworkDefinition</a></td><td class="desc">Abstracted representation of a network definition. This interface and the related types <a class="el" href="interface_sharp_neat_1_1_network_1_1_i_node_list.html" title="Represents a list of INetworkNode&#39;s. Part of the INetworkDefinition type hierarchy.">INodeList</a>, <a class="el" href="interface_sharp_neat_1_1_network_1_1_i_connection_list.html" title="Represents a list of INetworkConnection&#39;s. Part of the INetworkDefinition type hierarchy.">IConnectionList</a>, <a class="el" href="interface_sharp_neat_1_1_network_1_1_i_network_node.html" title="Represents a single network node. Part of the INetworkDefinition type hierarchy.">INetworkNode</a>, <a class="el" href="interface_sharp_neat_1_1_network_1_1_i_network_connection.html" title="Represents a single network connection. Part of the INetworkDefinition type hierarchy.">INetworkConnection</a>, etc, allow networks to be described abstractly </td></tr>
<tr id="row_0_5_17_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="interface_sharp_neat_1_1_network_1_1_i_network_node.html" target="_self">INetworkNode</a></td><td class="desc">Represents a single network node. Part of the <a class="el" href="interface_sharp_neat_1_1_network_1_1_i_network_definition.html" title="Abstracted representation of a network definition. This interface and the related types INodeList,...">INetworkDefinition</a> type hierarchy </td></tr>
<tr id="row_0_5_18_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="interface_sharp_neat_1_1_network_1_1_i_node_list.html" target="_self">INodeList</a></td><td class="desc">Represents a list of <a class="el" href="interface_sharp_neat_1_1_network_1_1_i_network_node.html" title="Represents a single network node. Part of the INetworkDefinition type hierarchy.">INetworkNode</a>'s. Part of the <a class="el" href="interface_sharp_neat_1_1_network_1_1_i_network_definition.html" title="Abstracted representation of a network definition. This interface and the related types INodeList,...">INetworkDefinition</a> type hierarchy </td></tr>
<tr id="row_0_5_19_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_network_1_1_leaky_re_l_u.html" target="_self">LeakyReLU</a></td><td class="desc">Leaky rectified linear activation unit (<a class="el" href="class_sharp_neat_1_1_network_1_1_re_l_u.html" title="Rectified linear activation unit (ReLU).">ReLU</a>) </td></tr>
<tr id="row_0_5_20_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_network_1_1_leaky_re_l_u_shifted.html" target="_self">LeakyReLUShifted</a></td><td class="desc">Leaky rectified linear activation unit (<a class="el" href="class_sharp_neat_1_1_network_1_1_re_l_u.html" title="Rectified linear activation unit (ReLU).">ReLU</a>). Shifted on the x-axis so that x=0 gives y=0.5, in keeping with the logistic sigmoid </td></tr>
<tr id="row_0_5_21_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_network_1_1_linear.html" target="_self">Linear</a></td><td class="desc"><a class="el" href="class_sharp_neat_1_1_network_1_1_linear.html" title="Linear activation function with clipping. By &#39;clipping&#39; we mean the output value is linear between x ...">Linear</a> activation function with clipping. By 'clipping' we mean the output value is linear between x = -1 and x = 1. Below -1 and above +1 the output is clipped at -1 and +1 respectively </td></tr>
<tr id="row_0_5_22_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_network_1_1_logistic_function.html" target="_self">LogisticFunction</a></td><td class="desc">The logistic function </td></tr>
<tr id="row_0_5_23_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_network_1_1_logistic_function_steep.html" target="_self">LogisticFunctionSteep</a></td><td class="desc">The logistic function with a steepened slope </td></tr>
<tr id="row_0_5_24_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_network_1_1_max_minus_one.html" target="_self">MaxMinusOne</a></td><td class="desc">y = max(-1, x). i.e. a rectified linear activation unit (<a class="el" href="class_sharp_neat_1_1_network_1_1_re_l_u.html" title="Rectified linear activation unit (ReLU).">ReLU</a>) variant </td></tr>
<tr id="row_0_5_25_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_network_1_1_network_connection.html" target="_self">NetworkConnection</a></td><td class="desc">Concrete implementation of <a class="el" href="interface_sharp_neat_1_1_network_1_1_i_network_connection.html" title="Represents a single network connection. Part of the INetworkDefinition type hierarchy.">INetworkConnection</a> </td></tr>
<tr id="row_0_5_26_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_network_1_1_network_connectivity_data.html" target="_self">NetworkConnectivityData</a></td><td class="desc"><a class="el" href="namespace_sharp_neat_1_1_network.html">Network</a> connectivity data for a network. Connectivity data in a form that is convenient for network traversal algorithms </td></tr>
<tr id="row_0_5_27_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_network_1_1_network_definition.html" target="_self">NetworkDefinition</a></td><td class="desc">Concrete implementation of <a class="el" href="interface_sharp_neat_1_1_network_1_1_i_network_definition.html" title="Abstracted representation of a network definition. This interface and the related types INodeList,...">INetworkDefinition</a>. This class represents network definitions independently of any other requirements, e.g. a NeatGenome is also an <a class="el" href="interface_sharp_neat_1_1_network_1_1_i_network_definition.html" title="Abstracted representation of a network definition. This interface and the related types INodeList,...">INetworkDefinition</a> but with additional baggage. This class is useful for representing network definitions, e.g. when creating a network instance from a HyperNEAT substrate </td></tr>
<tr id="row_0_5_28_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="struct_sharp_neat_1_1_network_1_1_network_depth_info.html" target="_self">NetworkDepthInfo</a></td><td class="desc">Conveys summary information from a network depth analysis </td></tr>
<tr id="row_0_5_29_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_network_1_1_network_node.html" target="_self">NetworkNode</a></td><td class="desc">Concrete implementation of an <a class="el" href="interface_sharp_neat_1_1_network_1_1_i_network_node.html" title="Represents a single network node. Part of the INetworkDefinition type hierarchy.">INetworkNode</a> </td></tr>
<tr id="row_0_5_30_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_network_1_1_node_connectivity_data.html" target="_self">NodeConnectivityData</a></td><td class="desc">Stored the IDs of nodes that connect in and out of a given node </td></tr>
<tr id="row_0_5_31_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_network_1_1_node_list.html" target="_self">NodeList</a></td><td class="desc">Concrete implementation of <a class="el" href="interface_sharp_neat_1_1_network_1_1_i_node_list.html" title="Represents a list of INetworkNode&#39;s. Part of the INetworkDefinition type hierarchy.">INodeList</a> </td></tr>
<tr id="row_0_5_32_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_network_1_1_null_fn.html" target="_self">NullFn</a></td><td class="desc">Null activation function. Returns zero regardless of input </td></tr>
<tr id="row_0_5_33_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_network_1_1_polynomial_approximant_steep.html" target="_self">PolynomialApproximantSteep</a></td><td class="desc">A very close approximation of the logistic function that avoids use of exp() and is therefore typically much faster to compute, while giving an almost identical sigmoid curve </td></tr>
<tr id="row_0_5_34_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_network_1_1_quadratic_sigmoid.html" target="_self">QuadraticSigmoid</a></td><td class="desc">A sigmoid formed by two sub-sections of the y=x^2 curve </td></tr>
<tr id="row_0_5_35_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_network_1_1_rbf_gaussian.html" target="_self">RbfGaussian</a></td><td class="desc"><a class="el" href="class_sharp_neat_1_1_network_1_1_gaussian.html" title="Gaussian activation function. Output range is 0 to 1, that is, the tails of the Gaussian distribution...">Gaussian</a> activation function. Output range is 0 to 1, that is, the tails of the <a class="el" href="class_sharp_neat_1_1_network_1_1_gaussian.html" title="Gaussian activation function. Output range is 0 to 1, that is, the tails of the Gaussian distribution...">Gaussian</a> distribution curve tend towards 0 as abs(x) -&gt; Infinity and the <a class="el" href="class_sharp_neat_1_1_network_1_1_gaussian.html" title="Gaussian activation function. Output range is 0 to 1, that is, the tails of the Gaussian distribution...">Gaussian</a> peak is at x = 0 </td></tr>
<tr id="row_0_5_36_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_network_1_1_re_l_u.html" target="_self">ReLU</a></td><td class="desc">Rectified linear activation unit (<a class="el" href="class_sharp_neat_1_1_network_1_1_re_l_u.html" title="Rectified linear activation unit (ReLU).">ReLU</a>) </td></tr>
<tr id="row_0_5_37_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_network_1_1_scaled_e_l_u.html" target="_self">ScaledELU</a></td><td class="desc">Scaled Exponential <a class="el" href="class_sharp_neat_1_1_network_1_1_linear.html" title="Linear activation function with clipping. By &#39;clipping&#39; we mean the output value is linear between x ...">Linear</a> Unit (SELU) </td></tr>
<tr id="row_0_5_38_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_network_1_1_sine.html" target="_self">Sine</a></td><td class="desc"><a class="el" href="class_sharp_neat_1_1_network_1_1_sine.html" title="Sine activation function with doubled period.">Sine</a> activation function with doubled period </td></tr>
<tr id="row_0_5_39_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_network_1_1_soft_sign_steep.html" target="_self">SoftSignSteep</a></td><td class="desc">The softsign sigmoid. This is a variant of softsign that has a steeper slope at and around the origin that is intended to be a similar slope to that of <a class="el" href="class_sharp_neat_1_1_network_1_1_logistic_function_steep.html" title="The logistic function with a steepened slope.">LogisticFunctionSteep</a> </td></tr>
<tr id="row_0_5_40_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_network_1_1_s_re_l_u.html" target="_self">SReLU</a></td><td class="desc">S-shaped rectified linear activation unit (<a class="el" href="class_sharp_neat_1_1_network_1_1_s_re_l_u.html" title="S-shaped rectified linear activation unit (SReLU). From: https://en.wikipedia.org/wiki/Activation_fun...">SReLU</a>). From: <a href="https://en.wikipedia.org/wiki/Activation_function">https://en.wikipedia.org/wiki/Activation_function</a> <a href="https://arxiv.org/abs/1512.07030">https://arxiv.org/abs/1512.07030</a> [Deep Learning with S-shaped Rectified <a class="el" href="class_sharp_neat_1_1_network_1_1_linear.html" title="Linear activation function with clipping. By &#39;clipping&#39; we mean the output value is linear between x ...">Linear</a> Activation Units] </td></tr>
<tr id="row_0_5_41_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_network_1_1_s_re_l_u_shifted.html" target="_self">SReLUShifted</a></td><td class="desc">S-shaped rectified linear activation unit (<a class="el" href="class_sharp_neat_1_1_network_1_1_s_re_l_u.html" title="S-shaped rectified linear activation unit (SReLU). From: https://en.wikipedia.org/wiki/Activation_fun...">SReLU</a>). Shifted on the x-axis so that x=0 gives y=0.5, in keeping with the logistic sigmoid. From: <a href="https://en.wikipedia.org/wiki/Activation_function">https://en.wikipedia.org/wiki/Activation_function</a> <a href="https://arxiv.org/abs/1512.07030">https://arxiv.org/abs/1512.07030</a> [Deep Learning with S-shaped Rectified <a class="el" href="class_sharp_neat_1_1_network_1_1_linear.html" title="Linear activation function with clipping. By &#39;clipping&#39; we mean the output value is linear between x ...">Linear</a> Activation Units] </td></tr>
<tr id="row_0_5_42_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_network_1_1_tan_h.html" target="_self">TanH</a></td><td class="desc"></td></tr>
<tr id="row_0_6_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_0_6_" class="arrow" onclick="toggleFolder('0_6_')">&#9658;</span><span class="icona"><span class="icon">N</span></span><a class="el" href="namespace_sharp_neat_1_1_phenomes.html" target="_self">Phenomes</a></td><td class="desc"></td></tr>
<tr id="row_0_6_0_" class="even" style="display:none;"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span id="arr_0_6_0_" class="arrow" onclick="toggleFolder('0_6_0_')">&#9658;</span><span class="icona"><span class="icon">N</span></span><a class="el" href="namespace_sharp_neat_1_1_phenomes_1_1_neural_nets.html" target="_self">NeuralNets</a></td><td class="desc"></td></tr>
<tr id="row_0_6_0_0_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_phenomes_1_1_neural_nets_1_1_connection.html" target="_self">Connection</a></td><td class="desc">Represents a single connection between two neurons in a <a class="el" href="class_sharp_neat_1_1_phenomes_1_1_neural_nets_1_1_cyclic_network.html" title="This class is provided for debugging and educational purposes. FastCyclicNetwork is functionally equi...">CyclicNetwork</a> </td></tr>
<tr id="row_0_6_0_1_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_phenomes_1_1_neural_nets_1_1_cyclic_network.html" target="_self">CyclicNetwork</a></td><td class="desc">This class is provided for debugging and educational purposes. <a class="el" href="class_sharp_neat_1_1_phenomes_1_1_neural_nets_1_1_fast_cyclic_network.html" title="A neural network class that represents a network with recurrent (cyclic) connections.">FastCyclicNetwork</a> is functionally equivalent and is much faster and therefore should be used instead of <a class="el" href="class_sharp_neat_1_1_phenomes_1_1_neural_nets_1_1_cyclic_network.html" title="This class is provided for debugging and educational purposes. FastCyclicNetwork is functionally equi...">CyclicNetwork</a> in most circumstances </td></tr>
<tr id="row_0_6_0_2_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_phenomes_1_1_neural_nets_1_1_fast_acyclic_network.html" target="_self">FastAcyclicNetwork</a></td><td class="desc">A neural network implementation for acyclic networks </td></tr>
<tr id="row_0_6_0_3_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="struct_sharp_neat_1_1_phenomes_1_1_neural_nets_1_1_fast_connection.html" target="_self">FastConnection</a></td><td class="desc">Working data struct for use in <a class="el" href="class_sharp_neat_1_1_phenomes_1_1_neural_nets_1_1_fast_cyclic_network.html" title="A neural network class that represents a network with recurrent (cyclic) connections.">FastCyclicNetwork</a> and sub-classes. Represents a single connection - its weight and source/target neurons </td></tr>
<tr id="row_0_6_0_4_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_phenomes_1_1_neural_nets_1_1_fast_cyclic_network.html" target="_self">FastCyclicNetwork</a></td><td class="desc">A neural network class that represents a network with recurrent (cyclic) connections </td></tr>
<tr id="row_0_6_0_5_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_phenomes_1_1_neural_nets_1_1_fast_relaxing_cyclic_network.html" target="_self">FastRelaxingCyclicNetwork</a></td><td class="desc">A version of <a class="el" href="class_sharp_neat_1_1_phenomes_1_1_neural_nets_1_1_fast_cyclic_network.html" title="A neural network class that represents a network with recurrent (cyclic) connections.">FastCyclicNetwork</a> that activates a network until it becomes 'relaxed' rather than for some fixed number of iterations. This class is exactly the same as <a class="el" href="class_sharp_neat_1_1_phenomes_1_1_neural_nets_1_1_fast_cyclic_network.html" title="A neural network class that represents a network with recurrent (cyclic) connections.">FastCyclicNetwork</a> in all other respects; See that class for more detailed info </td></tr>
<tr id="row_0_6_0_6_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="struct_sharp_neat_1_1_phenomes_1_1_neural_nets_1_1_layer_info.html" target="_self">LayerInfo</a></td><td class="desc">Stores a node and connection index that represent a layer within the network (the nodes and connections at a given depth in a network) </td></tr>
<tr id="row_0_6_0_7_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_phenomes_1_1_neural_nets_1_1_neuron.html" target="_self">Neuron</a></td><td class="desc">Represents a single neuron in a <a class="el" href="class_sharp_neat_1_1_phenomes_1_1_neural_nets_1_1_cyclic_network.html" title="This class is provided for debugging and educational purposes. FastCyclicNetwork is functionally equi...">CyclicNetwork</a> </td></tr>
<tr id="row_0_6_0_8_" class="even" style="display:none;"><td class="entry"><span style="width:64px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_phenomes_1_1_neural_nets_1_1_relaxing_cyclic_network.html" target="_self">RelaxingCyclicNetwork</a></td><td class="desc">A version of <a class="el" href="class_sharp_neat_1_1_phenomes_1_1_neural_nets_1_1_cyclic_network.html" title="This class is provided for debugging and educational purposes. FastCyclicNetwork is functionally equi...">CyclicNetwork</a> that activates a network until it becomes 'relaxed' rather than for some fixed number of iterations. This class is exactly the same as <a class="el" href="class_sharp_neat_1_1_phenomes_1_1_neural_nets_1_1_cyclic_network.html" title="This class is provided for debugging and educational purposes. FastCyclicNetwork is functionally equi...">CyclicNetwork</a> in all other respects; See that class for more detailed info </td></tr>
<tr id="row_0_6_1_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="interface_sharp_neat_1_1_phenomes_1_1_i_black_box.html" target="_self">IBlackBox</a></td><td class="desc"><a class="el" href="interface_sharp_neat_1_1_phenomes_1_1_i_black_box.html" title="IBlackBox represents an abstract device, system or function which has inputs and outputs....">IBlackBox</a> represents an abstract device, system or function which has inputs and outputs. The internal workings and state of the box are not relevant to any method or class that accepts an <a class="el" href="interface_sharp_neat_1_1_phenomes_1_1_i_black_box.html" title="IBlackBox represents an abstract device, system or function which has inputs and outputs....">IBlackBox</a> - only that it has inputs and outputs and a means of activation. In NEAT the neural network implementations generally fit this pattern, that is: </td></tr>
<tr id="row_0_6_2_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="interface_sharp_neat_1_1_phenomes_1_1_i_signal_array.html" target="_self">ISignalArray</a></td><td class="desc">An abstraction of the <a class="el" href="class_sharp_neat_1_1_phenomes_1_1_signal_array.html" title="SignalArray wraps a native array along with an offset into that array. The resulting SignalArray prov...">SignalArray</a> class. Provided to allow custom implementations of a signal array if required </td></tr>
<tr id="row_0_6_3_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_phenomes_1_1_mapping_signal_array.html" target="_self">MappingSignalArray</a></td><td class="desc"><a class="el" href="class_sharp_neat_1_1_phenomes_1_1_mapping_signal_array.html" title="MappingSignalArray wraps a native array along with an indirection/mapping array. See SignalArray for ...">MappingSignalArray</a> wraps a native array along with an indirection/mapping array. See <a class="el" href="class_sharp_neat_1_1_phenomes_1_1_signal_array.html" title="SignalArray wraps a native array along with an offset into that array. The resulting SignalArray prov...">SignalArray</a> for more info </td></tr>
<tr id="row_0_6_4_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_phenomes_1_1_output_mapping_signal_array.html" target="_self">OutputMappingSignalArray</a></td><td class="desc">A <a class="el" href="class_sharp_neat_1_1_phenomes_1_1_mapping_signal_array.html" title="MappingSignalArray wraps a native array along with an indirection/mapping array. See SignalArray for ...">MappingSignalArray</a> that applies the bounds interval [0,1] to returned values </td></tr>
<tr id="row_0_6_5_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_phenomes_1_1_output_signal_array.html" target="_self">OutputSignalArray</a></td><td class="desc">A <a class="el" href="class_sharp_neat_1_1_phenomes_1_1_signal_array.html" title="SignalArray wraps a native array along with an offset into that array. The resulting SignalArray prov...">SignalArray</a> that applies the bounds interval [0,1] to returned values </td></tr>
<tr id="row_0_6_6_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_phenomes_1_1_signal_array.html" target="_self">SignalArray</a></td><td class="desc"><a class="el" href="class_sharp_neat_1_1_phenomes_1_1_signal_array.html" title="SignalArray wraps a native array along with an offset into that array. The resulting SignalArray prov...">SignalArray</a> wraps a native array along with an offset into that array. The resulting <a class="el" href="class_sharp_neat_1_1_phenomes_1_1_signal_array.html" title="SignalArray wraps a native array along with an offset into that array. The resulting SignalArray prov...">SignalArray</a> provides offset indexed access to the underlying native array </td></tr>
<tr id="row_0_7_" class="even"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_0_7_" class="arrow" onclick="toggleFolder('0_7_')">&#9658;</span><span class="icona"><span class="icon">N</span></span><a class="el" href="namespace_sharp_neat_1_1_speciation_strategies.html" target="_self">SpeciationStrategies</a></td><td class="desc"></td></tr>
<tr id="row_0_7_0_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_speciation_strategies_1_1_k_means_clustering_strategy.html" target="_self">KMeansClusteringStrategy</a></td><td class="desc">An ISpeciationStrategy that speciates genomes using the k-means clustering method. k-means requires a distance metric and as such this class requires am IDistanceMetric to be provided at construction time. Different distance metrics can be used including NeatDistanceMetric which is equivalent to the metric used in the standard NEAT method albeit with a different clustering/speciation algorithm (Standard NEAT does not use k-means) </td></tr>
<tr id="row_0_7_1_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_speciation_strategies_1_1_parallel_k_means_clustering_strategy.html" target="_self">ParallelKMeansClusteringStrategy</a></td><td class="desc">An ISpeciationStrategy that speciates genomes using the k-means clustering method. k-means requires a distance metric and as such this class requires am IDistanceMetric to be provided at construction time. Different distance metrics can be used including NeatDistanceMetric which is equivalent to the metric used in the standard NEAT method albeit with a different clustering/speciation algorithm (Standard NEAT does not use k-means) </td></tr>
<tr id="row_0_7_2_" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_speciation_strategies_1_1_random_clustering_strategy.html" target="_self">RandomClusteringStrategy</a></td><td class="desc">A speciation strategy that allocates genomes to species randomly. Although allocation is random the strategy does maintain evenly sized species. Primarily used for testing/debugging and demonstrating comparative effectiveness of random allocation compared to other strategies </td></tr>
<tr id="row_0_8_"><td class="entry"><span style="width:16px;display:inline-block;">&#160;</span><span id="arr_0_8_" class="arrow" onclick="toggleFolder('0_8_')">&#9658;</span><span class="icona"><span class="icon">N</span></span><a class="el" href="namespace_sharp_neat_1_1_utility.html" target="_self">Utility</a></td><td class="desc"></td></tr>
<tr id="row_0_8_0_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="struct_sharp_neat_1_1_utility_1_1_point2_d_double.html" target="_self">Point2DDouble</a></td><td class="desc"></td></tr>
<tr id="row_0_8_1_" class="even" style="display:none;"><td class="entry"><span style="width:48px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_utility_1_1_u_int32_id_generator.html" target="_self">UInt32IdGenerator</a></td><td class="desc">Conveniently encapsulates a single UInt32, which is incremented to produce new IDs </td></tr>
<tr id="row_0_9_" class="even"><td class="entry"><span style="width:32px;display:inline-block;">&#160;</span><span class="icona"><span class="icon">C</span></span><a class="el" href="class_sharp_neat_1_1_sharp_neat_exception.html" target="_self">SharpNeatException</a></td><td class="desc">General purpose exception class for use within <a class="el" href="namespace_sharp_neat.html">SharpNeat</a> </td></tr>
</table>
</div><!-- directory -->
</div><!-- contents -->
</div><!-- doc-content -->
<!-- start footer part -->
<div id="nav-path" class="navpath"><!-- id is needed for treeview function! -->
  <ul>
    <li class="footer">Generated by
    <a href="http://www.doxygen.org/index.html">
    <img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.8.17 </li>
  </ul>
</div>
</body>
</html>
